src/shared/util.ts

```diff
@@ -161,4 +162,16 @@ export class Util {
   static trimIBAN(iban: string): string {
     return '***' + iban.slice(iban.length - 4);
   }
+
+  static transformToMinDeposit(deposit: any, filter?: string[] | string): MinDeposit[] {
+    return Object.keys(deposit)
+      .filter((key) => {
+        if (typeof filter === 'string') return filter === key;
+        if (typeof filter === 'object') return filter.includes(key);
+        return true;
+      })
+      .map((key) => {
+        return { amount: deposit[key], asset: key };
+      });
+  }
```

Comment:
that would change the behavior of the function as it would always filter, which is not wanted. But I do see some improvements I can make on your suggestion to the existing implementation.

The idea behind it was to have a customizable filter, which can be activated if needed - for example for staking routes. To only deliver min deposit for DFI

Reply:
Oh yes, you're completely right 😉